### DISCUSSION TITLE
message: "No such host is known."

### DISCUSSION THREAD

**Discussion by codebyaadi** [awslabs/aws-sdk-rust#1128] [General]
I am getting this error

`Object: Err(DispatchFailure(DispatchFailure { source: ConnectorError { kind: Io, source: hyper::Error(Connect, ConnectError("dns error", Os { code: 11001, kind: Uncategorized, message: "No such host is known." })), connection: Unknown } }))`

Below is my env
`AWS_ACCESS_KEY_ID=******************************
AWS_SECRET_ACCESS_KEY=*****************************************************
ENDPOINT=https://*****************************.r2.cloudflarestorage.com`

Below is my upload code

`use std::path::Path;

use aws_sdk_s3 as s3;
use aws_sdk_s3::error::SdkError;
use aws_sdk_s3::operation::put_object::{PutObjectError, PutObjectOutput};
use aws_sdk_s3::primitives::ByteStream;

pub async fn upload_file(
    file_name: &str,
    local_file_path: &str,
) -> Result<PutObjectOutput, SdkError<PutObjectError>> {
    let config = aws_config::load_from_env().await;
    let client = s3::Client::new(&config);

    let body = ByteStream::from_path(Path::new(local_file_path)).await;
    let request;
    match body {
        Ok(f) => {
            request = client
        .put_object()
        .bucket("vercel-clone")
        .key(file_name.to_string())
        .body(f)
        .send()
        .await?;
        },
        Err(e) => {
            panic!("Error uploading file: {:?}", e);
        }
    };

    println!("File uploaded successfully: {:?}", request);

    Ok(request)
}
`

This is where I am using it

`use axum::Json;
use git2::Repository;
use serde_json::{json, Value};
use std::env;

use crate::utils;
use crate::upload;

pub async fn home() -> Json<&'static str> {
    Json("Hello! World")
}

pub async fn deploy(Json(body): Json<Value>) -> Json<Value> {
     // some logic code
    let object = upload::upload_file("backend/output/txzil\\src\\mail_sender.rs", "D:\\MyWorks\\learning\\bunjs\\vercel-clone\\backend/output/txzil\\src\\mail_sender.rs").await;
    println!("Object: {:?}", object);
    Json(json!({"id": id, "files": files}))
}`

**Comment by ysaito1001** [comment]
Hi @codebyaadi, given the error message indicates "No such host is known.", can you ensure `https://*****************************.r2.cloudflarestorage.com` is reachable from the place where you execute your code? You could even verify that using AWS CLI to call [pub-object](https://docs.aws.amazon.com/cli/latest/reference/s3api/put-object.html#examples) instead of using AWS SDK for Rust.
